syscall.go

```diff
@@ -18,7 +18,11 @@ const maxArgs = 9
 //
 // On amd64, if there are more than 8 floats the 9th and so on will be placed incorrectly on the
 // stack.
-//go:nosplit
+//
+// the pragma go:nosplit is not needed at this function declaration because it uses go:uintptrescapes
+// which forces all the objects that the uintptrs point to onto the heap where a stack split won't affect
+// their memory location.
```

Comment:
So, my understanding is that the objects pointed by uintptrs are on the heap with fixed, not on the stack, and the objects' addresses never change during the call, and that's why we're safe without nosplit. Is that correct?

Reply:
If a pointer addresses to an object on the stack, what would happen? (A compiile error, or a runtime error?)

Reply:
I have confirmed that `uintptrespcapese` changes the escape analysis, so just having this annotation seems fine:

With `//go:uintptrescapes`

```go
package main

import (
        "unsafe"
)

//go:uintptrescapes
func Foo(x uintptr) {
}

func main() {
        var x int
        Foo(uintptr(unsafe.Pointer(&x)))
}
```

```
$ go run -gcflags="-m -N -l" escape.go 
# command-line-arguments
./escape.go:8:10: marking x as escaping uintptr
./escape.go:12:6: moved to heap: x
```

Without `//go:uintptrescapes`

```go
package main

import (
        "unsafe"
)

func Foo(x uintptr) {
}

func main() {
        var x int
        Foo(uintptr(unsafe.Pointer(&x)))
}
```

```
$ go run -gcflags="-m -N -l" escape.go
```

Reply:
You are correct as to how uintptrescapes works. If the pointer points to a stack object and there is no stack split then it is completely safe. If there is a stack split then the program can crash at runtime bc the C function might do just about anything with that memory which we don't know what its pointing to

Reply:
Thanks!